dhis-2/dhis-api/src/main/java/org/hisp/dhis/datavalue/DataValueStore.java

```diff
@@ -184,4 +185,12 @@ DataValue getDataValue(
    * @return true, if any value exist, otherwise false
    */
   boolean dataValueExists(CategoryCombo combo);
+
+  /**
+   * Checks if any data values exist for the provided {@link DataElement} {@link UID}.
+   *
+   * @param dataElementUid the {@link DataElement} {@link UID} to check
+   * @return true, if any values exist, otherwise false
+   */
+  boolean dataValueExistsForDataElement(String dataElementUid);
```

Comment:
Maybe we can just call the variable `uid` in these cases, it is implicit from the method name, and it makes the code a bit more elegant?